### DISCUSSION TITLE
LAB 05.01 Task 3: ML Metric vs Business Metric

### DISCUSSION THREAD

**Discussion by diegosaaval** [rramosp/ai4eng.v1#196] [05 - GESTIÓN DE MODELOS]
Saludos,

Felipe, intenté solucionarlo como le planteas en una respuesta anterior a un compañero, pero obtengo un error a la hora de calificar, agradecería mucho si revisas el código

Función de pnl:
```
def get_pnl(signal, predictions, trading_units=1):
    s = signal
    p = predictions
    
    converted_p = (p - 0.5) * 2
    pnl =  ( (s[1:] - s[:-1]) * trading_units ) * converted_p   -   s[:-1] * trading_units * 0.01
    return pnl
```
Matriz de 9x20:
```
pset = np.zeros( (9,20) ) #Matriz de ceros de 9x20
m = (np.linspace(0.1 , 1 , 10) * 20).astype(int)
for i in range(len(pset)):
    c = m[i] #Cantidad de unos
    k = np.random.permutation(20)[:c]
    pset[i,k] = pset[i,k] + 1
```
Salida:
![image](https://user-images.githubusercontent.com/106838105/187628962-c0f5a674-acd4-44a5-a36a-2ad2711a5397.png)

Itero en cada fila la función get_pnl:
```
pnl = np.round([get_pnl(s, pset[i]) for i in range(len(pset))], 2)` 

```
Salida:
![image](https://user-images.githubusercontent.com/106838105/187629304-b9b2b449-97f5-4a94-828c-fbc1ed19a741.png)

Error en autocorreción:
![image](https://user-images.githubusercontent.com/106838105/187627784-4c700815-e85f-4b96-9d41-e03609212aa5.png)

**Comment by Felipe-RA** [comment]
Saludos Diego

Este error se debe a que tu respuesta no tiene dimensiones correctas, el grader intenta comparar un arreglo de dimensiones `(9,)` con tu arreglo de dimensiones `(9,20)`

Recuerda que el `pnl` debe ser  un `array` o una `list` de nueve elementos (uno por cada fila de tu `PSET`), pero tu tienes un vector  de dimensiones`(9,20)`

Funciones recomendadas:

- `np.sum()` para obtener el `pnl` por fila en tu list comprehension

Cualquier duda, por aquí estamos!
